File: WEEK1/woohyun/신고 결과 받기.js
function solution(id_list, report, k) {
    var answer = [];
    
    var reporteeMap = {};   // key: 신고한 유저, value: 신고당한 유저 리스트
    var reporterMap = {};   // key: 신고당한 유저, value: 신고한 유저 리스트
    
    // 두 맵 모두 초기화
    for (id of id_list) { 
        reporteeMap[id] = [];
        reporterMap[id] = [];
    }
    

    for (_report of report) {
        // 신고 내역 리스트 split (신고자-피신고자)
        let splitted = _report.split(" ");
        let reporter = splitted[0];
        let reportee = splitted[1];
        
        // 신고자 맵에 중복 검사 후 저장
        if(!reporteeMap[reporter].includes(reportee)) {
            reporteeMap[reporter].push(reportee);
        }
        
        // 피신고자 맵에 중복 검사 후 저장
        if(!reporterMap[reportee].includes(reporter)) {
            reporterMap[reportee].push(reporter);
        }
    }
    
    // 신고자 맵에서 피신고자 리스트를 탐색한 뒤 신고당한 횟수 비교 후 정답 출력
    for (id of id_list) {
        var count = 0;
        
        for (reportee of reporteeMap[id]) {
            if(reporterMap[reportee].length >= k) count++;
        }
        
        answer.push(count);
    }
    
    return answer;
}

File: WEEK1/yuncheol/신고 결과 받기.java
import java.util.*;

class Solution {
    public int[] solution(String[] id_list, String[] report, int k) {
        int id_count = id_list.length;
        int report_count = report.length;
        
        int[] answer = new int[id_count];
        
        Map<String, List<String>> report_map = new HashMap<>();
        Map<String, Integer> delete_map = new HashMap<>();
        Map<String, Integer> answer_map = new HashMap<>();
        
        report = Arrays.stream(report).distinct().toArray(String[]::new);
        
        for (int i = 0; i < id_count; i++) {
            report_map.put(id_list[i], new ArrayList<String>());
            delete_map.put(id_list[i], k);
            answer_map.put(id_list[i], 0);
        }
        
        for (String rpt: report) {
            String[] from_to = rpt.split(" ");
            String from = from_to[0];
            String to = from_to[1];
            
            report_map.get(from).add(to);
            delete_map.put(to, delete_map.get(to) - 1);
        }
        
        for (String key: answer_map.keySet()) {
            for (String report_value: report_map.get(key)) {
                if (delete_map.get(report_value) <= 0) {
                    answer_map.put(key, answer_map.get(key) + 1);
                }
            }
        }
        
        for (int i = 0; i < id_count; i++) {
            answer[i] = answer_map.get(id_list[i]);
        }
        
        return answer;
    }
}